src/h3/_cy/geo.pyx

```diff
@@ -0,0 +1,270 @@
+
+    n = h3lib.maxPolyfillSize(&gp.gp, res)
+    ptr = create_ptr(n)
+
+    h3lib.polyfill(&gp.gp, res, ptr)
+    mv = create_mv(ptr, n)
+
+    return mv
+
+
+def polyfill_geojson(geojson, int res):
+    """ Set of hexagons whose center is contained in a GeoJson Polygon object.
+
+    The polygon is defined exactly as in the GeoJson standard, so
+    `geojson` should be a dictionary like:
+    {
+        'type': 'Polygon',
+        'coordinates': [...]
+    }
+
+    'coordinates' should be a list of LinearRings, where the first ring describes
+    the exterior boundary of the Polygon, and any subsequent LinearRings
+    describe holes in the polygon.
+
+    Note that we don't provide an option for the order of the coordinates,
+    as the GeoJson standard requires them to be in lng/lat order.
+
+    Parameters
+    ----------
+    geojson : dict
+    res : int
+        The resolution of the output hexagons
+    """
+
+    # todo: this one could handle multipolygons...
+
+    if geojson['type'] != 'Polygon':
+        raise ValueError('Only Polygon GeoJSON supported')
+
+    coords = geojson['coordinates']
+
+    out = polyfill_polygon(coords[0], res, holes=coords[1:], lnglat_order=True)
+
+    return out
+
+
+def polyfill(dict geojson, int res, bool geo_json_conformant=False):
+    """ Light wrapper around `polyfill_geojson` to provide backward compatibility.
+    """
+
+    try:
+        gj_type = geojson['type']
+    except KeyError:
+        raise KeyError("`geojson` dict must have key 'type'.") from None
+
+    if gj_type != 'Polygon':
+        raise ValueError('Only Polygon GeoJSON supported')
+
+    if geo_json_conformant:
+        out = polyfill_geojson(geojson, res)
+    else:
+        coords = geojson['coordinates']
+        out = polyfill_polygon(coords[0], res, holes=coords[1:], lnglat_order=False)
+
+    return out
+
+
+def cell_boundary(H3int h, bool geo_json=False):
+    """Compose an array of geo-coordinates that outlines a hexagonal cell"""
+    cdef:
+        h3lib.GeoBoundary gb
+
+    check_cell(h)
+
+    h3lib.h3ToGeoBoundary(h, &gb)
+
+    verts = tuple(
+        coord2deg(gb.verts[i])
+        for i in range(gb.num_verts)
+    )
+
+    if geo_json:
+        #lat/lng -> lng/lat and last point same as first
+        verts = tuple(tuple(reversed(v)) for v in verts)
+        verts += (verts[0],)
+
+    return verts
+
+
+def edge_boundary(H3int edge, bool geo_json=False):
+    """ Returns the GeoBoundary containing the coordinates of the edge
+    """
+    cdef:
+        h3lib.GeoBoundary gb
+
+    check_edge(edge)
+
+    h3lib.getH3UnidirectionalEdgeBoundary(edge, &gb)
+
+    # todo: move this verts transform into the GeoBoundary object
+    verts = tuple(
+        coord2deg(gb.verts[i])
+        for i in range(gb.num_verts)
+    )
+
+    if geo_json:
+        #lat/lng -> lng/lat and last point same as first
+        verts = tuple(tuple(reversed(v)) for v in verts)
+        verts += (verts[0],)
```

Comment:
If there's not a performance difference, could we use `tuple([verts[0]])`? This use of the comma operator seems somewhat tricky if you're not familiar with what it's doing.

Reply:
I'd actually say that the `(verts[0],)` is [pretty idiomatic python](https://stackoverflow.com/questions/1380860/add-variables-to-tuple).

That said, I did refactor a bit. Let me know what you think.